prefect_transform/credentials.py

```diff
@@ -0,0 +1,77 @@
+"""Transform credentials block"""
+import os
+from typing import Optional
+
+from prefect.blocks.core import Block
+from pydantic import SecretStr, root_validator
+from transform import MQLClient
+from transform.exceptions import AuthException, URLException
+
+from prefect_transform.exceptions import (
+    TransformAuthException,
+    TransformConfigurationException,
+)
+
+
+class TransformCredentials(Block):
+    """
+    Block used to manage authentication with Transform.
+    """
+
+    api_key: Optional[SecretStr] = None
+    api_key_env_var: Optional[str] = None
```

Comment:
Wondering if we should remove these `env_var` fields since they aren't portable across machines?